llvm/lib/Target/AArch64/GISel/AArch64InstructionSelector.cpp

```diff
@@ -2045,7 +2045,8 @@ bool AArch64InstructionSelector::selectVaStartAAPCS(
                   (*I.memoperands_begin())
                       ->getPointerInfo()
                       .getWithOffset(OffsetBytes),
-                  MachineMemOperand::MOStore, PtrSize, Align(PtrSize)));
+                  MachineMemOperand::MOStore, PtrSize,
+                  commonAlignment(Align(PtrSize), OffsetBytes)));
```

Comment:
Actually I was wrong, the MMO alignment is a function of commonAlignment(BaseAlign, Offset), so this should be passing through the same base alignment from MMO->getBaseAlign()